src/core/atomic.d

```diff
@@ -539,9 +666,9 @@ private
     }
 
     // TODO: it'd be nice if we had @trusted scopes; we could remove this...
```

Comment:
Could be replaced with a `@trusted` lambda (something like `(() @trusted => casWeak(&val, &get, set))()`)? Not sure if that's better here though.

Reply:
Yeah, possibly the lamest pattern D ever manifested. I don't want to put another another another function call around an asm opcode. There's already 3-4 more than enough.